tests/test_adjoint.py

```diff
@@ -33,36 +33,30 @@
 device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
 dtype = torch.get_default_dtype()
 
-ito_methods = {'euler': 'ito',
-               'milstein': 'ito',
-               'srk': 'ito'}
-stratonovich_methods = {'midpoint': 'stratonovich',
-                        'log_ode': 'stratonovich'}
+ito_methods = {'milstein': 'ito', 'srk': 'ito'}
+stratonovich_methods = {'midpoint': 'stratonovich', 'log_ode': 'stratonovich'}
 
 
 @pytest.mark.parametrize("problem", [Ex1, Ex2, Ex3])
 @pytest.mark.parametrize("method, sde_type", itertools.chain(ito_methods.items(), stratonovich_methods.items()))
 @pytest.mark.parametrize("noise_type", ['diagonal', 'scalar', 'additive', 'general'])
 @pytest.mark.parametrize('adaptive', (False, True))
 def test_adjoint(problem, method, sde_type, noise_type, adaptive):
-    if method == 'euler' and adaptive:
-        return
     if problem is not Ex3 and noise_type == 'additive':
         return
-    # TODO: remove this once we have adjoint implemented for other noise/sde combinations
-    if sde_type == 'stratonovich' and noise_type != 'general':
-        return
     if sde_type == 'ito' and noise_type == 'general':
         return
+    if noise_type == "diagonal" and method == "log_ode":
+        return
```

Comment:
So I get that log_ode is the same as midpoint when the noise is non-general. If that's the reason for this check then all non-general noise_types should trigger this.

That said, I'd prefer to test log-ode anyway. It is a valid combination.